test/app.test.ts

```diff
@@ -12,22 +12,27 @@ DataLakeService.prototype.listFiles = jest
 
 DataLakeService.prototype.uploadFile = jest.fn();
 
-beforeAll(async () => {
-    await connectToDb(datasourceOptions);
-    const datafile1 = new Datafile();
-    datafile1.name = 'test-data-1.csv';
-    datafile1.description = 'Test Data File 1';
-    datafile1.id = 'bdc40218-af89-424b-b86e-d21710bc92f1';
-    await datafile1.save();
-    const datafile2 = new Datafile();
-    datafile2.name = 'test-data-2.csv';
-    datafile2.description = 'Test Data File 2';
-    datafile2.id = 'fa07be9d-3495-432d-8c1f-d0fc6daae359';
-    await datafile2.save();
-    console.log(`Datafile created: ${Datafile.find()}`);
-});
-
 describe('Test app.ts', () => {
+    beforeAll(async () => {
+        await connectToDb(datasourceOptions);
+        await dbManager.initializeDataSource();
+
+        const dataset1 = Dataset.createDataset('Test Data 1', 'test', 'bdc40218-af89-424b-b86e-d21710bc92f1');
```

Comment:
Add a test for database-assigned UUIDs if you're going to keep them in the schema.

Reply:
I can do this...  my hesitation is that this test suite substitutes BetterSQLite3 for Postgres and this feels like something which isn't in my control... If the database fails to get a UUID automatically then I have other issues.  Also based on your previous comments we might switch to creating UUIDs ourselves in code rather than relying on the database to do it for us.